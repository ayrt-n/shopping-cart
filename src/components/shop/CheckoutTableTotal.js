import React from 'react';

function CheckoutTableTotal({ cart }) {
  const total = cart.reduce((sum, item) => sum + (item.quantity * item.price), 0);

  return (
    <table>
      <tbody>
        <tr>
          <th colSpan="2">Total</th>
          <td data-testid="checkout-total">${total}</td>
        </tr>
      </tbody>
    </table>
  );
}

export default CheckoutTableTotal;
