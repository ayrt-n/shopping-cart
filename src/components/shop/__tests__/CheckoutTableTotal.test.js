import React from 'react';
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import CheckoutTableTotal from '../CheckoutTableTotal';

describe('CheckoutTableTotal component', () => {
  it('renders table row correctly', () => {
    const { container } = render(<CheckoutTableTotal cart={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('calculates the total and displays in table row', () => {
    const cart = [
      {
        quantity: 2,
        price: 2,
      },
      {
        quantity: 5,
        price: 5,
      },
    ]

    render(<CheckoutTableTotal cart={cart} />)
    const tableTotal = screen.getByTestId('checkout-total');

    expect(tableTotal).toHaveTextContent('$29');
  });
});