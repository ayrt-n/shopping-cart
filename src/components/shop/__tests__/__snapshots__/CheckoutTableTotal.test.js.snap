// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutTableTotal component renders table row correctly 1`] = `
<div>
  <tr>
    <th
      colspan="2"
    >
      Total
    </th>
    <td
      data-testid="checkout-total"
    >
      $
      0
    </td>
  </tr>
</div>
`;
